add tutorial guidelines, remove concepts/architecture (#1258)
marin-community/marin: docs/explanation/tutorial-guidelines.md

This document describes a checklist for writing tutorials.

- Write in the second person (i.e., "you" instead of "we").
- Start with a high-level conceptual overview of what the user will accomplish.

> - There should a "Prerequisites" section that lists the software requirements as well as other tutorials that the user should complete prior to starting the current tutorial.

Reviewer: ```suggestion
- There should be a "Prerequisites" section that lists the software requirements as well as other tutorials that the user should complete prior to starting the current tutorial.
```

Reviewer: this makes it sound like every tutorial should say to make sure you have python installed in the prereqs...

```suggestion
- There should be a "Prerequisites" section that links to tutorials that the user should complete prior to starting the current tutorial as well as any additional software needed for this tutorial
```

- Do not include steps that are redundant with prerequisites (e.g., installing Python).
- Write down explicitly the resource requirements for the tutorial (e.g., whether GPUs are required).
- Specify the approximate time it takes to complete the tutorial (both in human time and in run time).

> - Walk through a concrete example that the user should be able to run fully (commands whould have concrete arguments that can be copy/pasted, not filled with placeholders that the user has to think about).

Reviewer: ```suggestion
- Walk through a concrete example that the user should be able to run fully (unless impossible, commands should have concrete arguments that can be copy/pasted, not filled with placeholders that the user has to think about).
```
